File: front-ui-vue/src/api/userBasic/userBasic.ts

import request from '@/utils/request'

export function createUserBasic(data){
    return request({
        url: '/api/app/user/add',
        method: 'post',
        data: data,
    })
}
export function login(data){
    return request({
        url: '/api/app/user/login',
        method: 'post',
        data: data,
    })
}

export function detail(id: string) {
    return request({
        url: '/api/app/user/detail/' + id,
        method: 'post'
    })
}
export function logout() {
    return request({
        url:"/api/app/user/logout",
        method: 'post',
    })
}

export function userSearch(data) {
    return request({
        url:"/api/app/user/search",
        method:"post",
        data: data,
    })
}

File: server-go/internal/app/router/user_basic.go
package router

import (
	"github.com/gin-gonic/gin"
	"server-go/internal/app/control"
	"server-go/internal/app/middleware"
)

func UserBasicRouter(r *gin.RouterGroup) {
	user := r.Group("user")
	user.POST("add", control.UserControl.CreateUserBasic)                                     // 新增用户
	user.POST("update", middleware.AuthMiddlewareUpdate, control.UserControl.UpdateUserBasic) // 修改用户信息
	user.POST("delete", control.UserControl.DeleteUserBasic)                                  // 删除用户信息
	user.POST("login", control.UserControl.Login)                                             // 登录
	user.POST("logout", middleware.AuthMiddlewareUpdate, control.UserControl.Logout)          // 退出登录
	user.POST("detail/:id", middleware.AuthMiddlewareUpdate, control.UserControl.UserDetail)  // 查看用户信息
	user.POST("search", middleware.AuthMiddlewareUpdate, control.UserControl.SearchUser)      // 通过 id or email or phone 查询用户
}
